### PR TITLE
Convert ComCtl32 ImageList Methods to Cswin32

### PR DESCRIPTION
**Pull request by lonitra**
- Convert ImageList methods: `Add`,`Create`,`Destroy`,`Draw`,`DrawEx`,`Duplicate`,`GetIconSize`,`GetImageCount`,`GetImageInfo`,`Remove`,`Replace`,
`ReplaceIcon`,`SetBkColor`
- Removed enum/struct: `ILC`,`IMAGEINFO`

Note:
`ILD` enum has not been removed due to [missing enum values in metadata](https://github.com/microsoft/win32metadata/issues/1285) 
ImageList `Read`/`Write`/`WriteEx` method conversion postponed until COM work done on cswin32 side

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/7866)

### CODE REVIEW

**Review by dreddy-work — APPROVED** [review]
LGTM.

**Review by JeremyKuhne — CHANGES_REQUESTED** [review]
Nice! Just a few comments.

File: src/System.Windows.Forms.Primitives/src/Windows/Win32/PInvoke.ImageList.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            {
+                bool result = ImageList_Remove(himl.Handle, i);
+                GC.KeepAlive(himl.Wrapper);
+                return result;
+            }
+
+            public static bool Replace<T>(in T himl, int i, HBITMAP hbmImage, HBITMAP hbmMask) where T : IHandle<HIMAGELIST>
+            {
+                bool result = ImageList_Replace(himl.Handle, i, hbmImage, hbmMask);
+                GC.KeepAlive(himl);
```
[the remark this answers is not in the document]
Comment (lonitra) [reply src/System.Windows.Forms.Primitives/src/Windows/Win32/PInvoke.ImageList.cs:86]: Thanks, missed this!

File: src/System.Windows.Forms/src/System/Windows/Forms/ImageList.NativeImageList.cs [hunk 2]
```diff
@@ -7,13 +7,12 @@
...
 using System.Drawing;
 using static Interop;
-using static Interop.ComCtl32;
 
 namespace System.Windows.Forms
 {
     public sealed partial class ImageList
     {
-        internal class NativeImageList : IDisposable, IHandle
+        internal class NativeImageList : IDisposable, IHandle, IHandle<HIMAGELIST>
```
[the remark this answers is not in the document]
Comment (lonitra) [reply src/System.Windows.Forms/src/System/Windows/Forms/ImageList.NativeImageList.cs:15]: Ah, yes this can actually be removed here.

**Review by JeremyKuhne — APPROVED** [review]
:shipit:

File: src/System.Windows.Forms.Primitives/src/Windows/Win32/PInvoke.ImageList.cs [hunk 3]
```diff
@@ -0,0 +1,109 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Windows.Win32
+{
+    internal partial class PInvoke
+    {
+        public static class ImageList
```
Comment (RussKie) [inline src/System.Windows.Forms.Primitives/src/Windows/Win32/PInvoke.ImageList.cs:9]: ❤️ for the alphabetical order of members

**Pull request merged**